pkg/dnsutil/hostsstore/hostsstore.go

```diff
@@ -114,6 +114,9 @@ func (x *hostsStore) Acquire(meta Meta) (err error) {
 		if err = os.WriteFile(loc, []byte{}, 0o644); err != nil {
 			return errors.Join(store.ErrSystemFailure, err)
 		}
+		if err = os.Chmod(loc, 0o644); err != nil {
```

Comment:
Could you add a comment to explain why `0o644` in the arg of `WriteFile` doesn't take an expected effect?

Reply:
Done